lib/hotel_booker.rb

```diff
@@ -0,0 +1,117 @@
+require 'pry'
+require_relative 'room'
+
+
+module Hotel
+  
+  class HotelBooker
+    
+    attr_accessor :rooms, :reservations, :blocks
+    
+    def initialize(num_of_rooms)
+      rooms = Hotel::Room.load_all(num_of_rooms)
+      @rooms = rooms
+      @reservations = []
+      @blocks = []
+      
+    end
+    
+    def print_rooms
+      list = ""
+      @rooms.each do |room|
+        list += "Room #{room.number}""\n"
+      end
+      return list
+    end
+    
+    def new_date_range(check_in, check_out)
+      date_range = Hotel::DateRange.new(check_in, check_out)
+    end
+    
+    def new_reservation(check_in, check_out)
+      
+      date_range = self.new_date_range(check_in, check_out)
+      avail_rooms = self.available_rooms(check_in, check_out)
+      
+      selected_room = avail_rooms[0]
+      new_reservation = Hotel::Reservation.new(selected_room, date_range)
+      
+      @reservations << new_reservation
+      
+    end
+    
+    
+    def available_rooms(check_in, check_out)
+      
+      same_date_res = @reservations.select do |reservation| 
+        reservation.date_range.overlap?(check_in, check_out) == true
+      end
+      
+      @blocks.each do |block|
+        block.reservations.each do |reservation|
+          if reservation.date_range.overlap?(check_in, check_out) == true
+            same_date_res << reservation
+          end
+        end
+      end
+      
+      taken_rooms = same_date_res.map { |reservation| reservation.room }
+      
+      avail_rooms = @rooms.select { |room| taken_rooms.include?(room) == false }
+      
+      if avail_rooms.length == 0
+        raise ArgumentError.new("There are no hotel rooms avaialbe for that time frame")
```

Comment:
This should probably be a `StandardError` or a custom error since it's raised because of the state of your program, not the arguments themselves.  

(Most of our examples use `ArgumentError` so this was a pretty common mistake and we'll clarify the point for next cohort.)